test/uk/gov/hmrc/excisemovementcontrolsystemapi/connectors/EISSubmissionConnectorSpec.scala

```diff
@@ -132,94 +139,85 @@ class EISSubmissionConnectorSpec
       clean(captor.value) mustBe clean(controlWrappedXml.toString)
     }
 
-    "return Bad request error" in {
+    "return error when a bad request response returned" in {
 
       when(mockHttpClient.post(any)(any)).thenReturn(mockRequestBuilder)
       when(mockRequestBuilder.setHeader(any())).thenReturn(mockRequestBuilder)
       when(mockRequestBuilder.withBody(any())(any(), any(), any())).thenReturn(mockRequestBuilder)
-      when(mockRequestBuilder.execute[Either[Result, EISSubmissionResponse]](any(), any()))
-        .thenReturn(Future.successful(Left(BadRequest("any error"))))
+      when(mockResponse.json).thenReturn(Json.toJson(EISErrorResponse(timestamp, "400", "", "", "")))
+      when(mockResponse.status).thenReturn(BAD_REQUEST)
 
       val result = await(submitExciseMovementForIE815)
-      result.left.value mustBe BadRequest("any error")
+      result.left.value mustBe EISErrorResponseDetails(BAD_REQUEST, timestamp, "", "", "", None)
     }
 
-    "return 500 if post request fail" in {
+    //Should this instead be a Future.failed?
+    "return EISErrorResponseDetails outlining a 500 if post request fail" in {
       when(mockHttpClient.post(any)(any)).thenReturn(mockRequestBuilder)
       when(mockRequestBuilder.setHeader(any())).thenReturn(mockRequestBuilder)
       when(mockRequestBuilder.withBody(any())(any(), any(), any())).thenReturn(mockRequestBuilder)
-      when(mockRequestBuilder.execute[Either[Result, EISSubmissionResponse]](any(), any()))
+      when(mockRequestBuilder.execute[HttpResponse](any(), any()))
         .thenReturn(Future.failed(new RuntimeException("error")))
 
       val result = await(submitExciseMovementForIE815)
 
-      result.left.value mustBe InternalServerError(
-        Json.toJson(
-          EisErrorResponsePresentation(
-            timestamp,
-            "Internal server error",
-            "Unexpected error occurred while processing Submission request",
-            emcsCorrelationId
-          )
-        )
+      result.left.value mustBe EISErrorResponseDetails(
+        INTERNAL_SERVER_ERROR,
+        timestamp,
+        "Internal server error",
+        "Unexpected error occurred while processing Submission request",
+        emcsCorrelationId
       )
     }
 
-    "return Not found error" in {
-
-      when(mockHttpClient.post(any)(any)).thenReturn(mockRequestBuilder)
-      when(mockRequestBuilder.setHeader(any())).thenReturn(mockRequestBuilder)
-      when(mockRequestBuilder.withBody(any())(any(), any(), any())).thenReturn(mockRequestBuilder)
-      when(mockRequestBuilder.execute[Either[Result, EISSubmissionResponse]](any(), any()))
-        .thenReturn(Future.successful(Left(ServiceUnavailable("any error"))))
-
-      val result = await(submitExciseMovementForIE815)
-
-      result.left.value mustBe ServiceUnavailable("any error")
-    }
+    //Can probably refactor a number of these to a single iterative test
```

Comment:
Will this refactor follow?

Reply:
This will follow in a future PR for this ticket, yes.